### PR TITLE
Donor signs up with supported zipcode

### PR DESCRIPTION
**Pull request by seanpdoyle**
https://trello.com/c/DcpszTGS

This commit introduces the [`clearance`][clearance] gem to handle our
`User` model.

When registering, donors must provide:

* an email address
* their primary zipcode
* a password

Once registered, we'll explain to those outside of our supported Zipcode
that they'll be notified once Fresh Food Connect is rolled out to their
locations.

For those within our supported areas, we'll redirect them to their
profiles so that they can manage their pickup locations.

[clearance]: https://github.com/thoughtbot/clearance

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -4,6 +4,7 @@ ruby "2.3.0"
 
 gem "autoprefixer-rails"
 gem "bourbon", "5.0.0.beta.3"
+gem "clearance"
```
Comment (geoffharcourt) [inline Gemfile]: You might want to lock to the 1.x series before the 2.x series gets released some time this year.

File: spec/factories.rb [hunk 2]
```diff
@@ -1,2 +1,7 @@
 FactoryGirl.define do
+  factory :user do
+    sequence(:email) { |i| "user#{i}@example.com" }
+    password "password"
+    zipcode { "80221" }
```
Comment (geoffharcourt) [inline spec/factories.rb]: What does making this attribute dynamic do?

File: spec/features/guest_views_unautorized_page_spec.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  scenario "they're redirected to the sign in page" do
+    visit profile_path
+
+    expect(page).to have_sign_in_text
+  end
+
+  def have_sign_in_text
+    have_text t("helpers.submit.session.submit")
+  end
+end
```
Comment (geoffharcourt) [inline spec/features/guest_views_unautorized_page_spec.rb]: The cleanest Capybara spec ever.

File: spec/factories.rb (hunk 2 above)
Comment (seanpdoyle) [inline spec/factories.rb]: Do you mean to ask why it's lazy?

The answer is "I don't know, it doesn't need to be"

**Pull request merged**